File: 2025-2/Clase24/STL_VECTOR/src/Persona.cpp
//
// Created by Erasmo on 5/11/25.
//

#include "Persona.hpp"

Persona::Persona() {
}

Persona::Persona(const Persona &orig) {
    *this = orig;
}

Persona::~Persona() {
}

void Persona::SetSueldo(double sueldo) {
    this->sueldo = sueldo;
}

double Persona::GetSueldo() const {
    return sueldo;
}

void Persona::SetDni(int dni) {
    this->dni = dni;
}

int Persona::GetDni() const {
    return dni;
}

void Persona::SetNombre(string nombre) {
    this->nombre = nombre;
}

string Persona::GetNombre() const {
    return nombre;
}

void Persona::leer() {
    char buffer[30];
    cin>>dni;
    if(cin.eof())return;
    cin.get();
    cin.getline(buffer, 30, ',');
    nombre = buffer; //Esta sobrecarga ya se encuentra en la clase String
    //SetNombre(buffer);
    cin>>sueldo;
}

void Persona::leer(ifstream & input) {
    char buffer[50];
    input>>dni;
    if(input.eof())return;
    input.get();
    input.getline(buffer, 50, ',');
    nombre = buffer; //Esta sobrecarga ya se encuentra en la clase String
    //SetNombre(buffer);
    input>>sueldo;
}

void Persona::imprimir() const {
    cout.precision(2);
    cout << fixed;
    cout << right << setw(10) << dni << " " << left << setw(30) << nombre;
    cout << right << setw(10) << sueldo << endl;
}

void Persona::imprimir(ofstream& output) const {
    output.precision(2);
    output << fixed;
    output << right << setw(10) << dni << " " << left << setw(30) << nombre;
    output << right << setw(10) << sueldo << endl;
}

bool Persona::operator<(const Persona &p) const {
    return this->sueldo<p.sueldo;
}

void Persona::operator=(const Persona &p) {
    this->sueldo = p.sueldo;
    this->nombre = p.nombre;
    this->dni = p.dni;
}

void operator>>(istream &input, class Persona &p) {
    p.leer();
}

ostream &operator<<(ostream &output, const class Persona &p) {
    p.imprimir();
    return output;
}

void operator>>(ifstream &input, class Persona &p) {
    p.leer(input);
}

ofstream &operator<<(ofstream &output, const class Persona &p) {
    p.imprimir(output);
    return output;
}

File: 2025-2/Clase26/JavaSegundoProyecto/SegundoProyecto/src/Persona.java
public class Persona {
    public int getDni() {
        return dni;
    }

    public void setDni(int dni) {
        this.dni = dni;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getApellido() {
        return apellido;
    }

    public void setApellido(String apellido) {
        this.apellido = apellido;
    }

    public double getSueldo() {
        return sueldo;
    }

    public void setSueldo(double sueldo) {
        this.sueldo = sueldo;
    }

    public boolean isEstado() {
        return estado;
    }

    public void setEstado(boolean estado) {
        this.estado = estado;
    }

    public char getSexo() {
        return sexo;
    }

    public void setSexo(char sexo) {
        this.sexo = sexo;
    }

    public Persona() {
        sueldo = 0.0;
        estado = false;
    }

    public Persona(int dni, String nombre, String apellido, double sueldo, boolean estado, char sexo) {
        this.dni = dni;
        this.nombre = nombre;
        this.apellido = apellido;
        this.sueldo = sueldo;
        this.estado = estado;
        this.sexo = sexo;
    }

    public void imprimir(){
        System.out.println("DNI: " + dni);
        System.out.println("Nombre: " + nombre);
        System.out.println("Apellido: " + apellido);
        System.out.println("Sueldo: " + sueldo);
        System.out.println("Estado: " + estado);
        System.out.println("Sexo: " + sexo);
    }

    private int dni;
    private String nombre;
    private String apellido;
    private double sueldo;
    private boolean estado;
    private char sexo;


}
